tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/wizard/AbstractLiferayWizard.java

```diff
@@ -0,0 +1,100 @@
+/*******************************************************************************
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+
+package com.liferay.ide.project.ui.wizard;
+
+import com.liferay.ide.core.util.CoreUtil;
+import com.liferay.ide.ui.LiferayPerspectiveFactory;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.IConfigurationElement;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.sapphire.Element;
+import org.eclipse.sapphire.ui.def.DefinitionLoader;
+import org.eclipse.sapphire.ui.forms.FormComponentPart;
+import org.eclipse.sapphire.ui.forms.WizardDef;
+import org.eclipse.sapphire.ui.forms.swt.SapphireWizard;
+import org.eclipse.ui.INewWizard;
+import org.eclipse.ui.IWorkbench;
+import org.eclipse.ui.IWorkbenchWizard;
+import org.eclipse.ui.IWorkingSet;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.wizards.newresource.BasicNewProjectResourceWizard;
+import org.eclipse.ui.wizards.newresource.BasicNewResourceWizard;
+import org.eclipse.wst.web.internal.DelegateConfigurationElement;
+
+/**
+ * @author Terry Jia
+ */
+@SuppressWarnings( "restriction" )
+public class AbstractLiferayWizard<T extends Element> extends SapphireWizard implements IWorkbenchWizard, INewWizard
```

Comment:
You are creating an AbstractLiferayWizard but only one other class is sub-classes that one, the new LiferayWorkspace wizard.  Just merge these two classes, unless you have a 2nd class that needs to sub-class the Abstract One.

In general... don't prematurely optimize things... don't make an Abstract class if there is only one single sub-class.  Just simply make everything in one class.  Only introduce an abstract class if there are two sub-classes.